### ISSUE TITLE
As WRES team, I would like to identify a candidate for github Release v6.26, test it, and if it passes, deploy it; candidate RC1: 80977b3e4126248585bff3c800c0866a446881ff wres-20240913-80977b3.zip

### ISSUE THREAD

**Issue by HankHerr-NOAA** [NOAA-OWP/wres#312]
- [x] 1 - Update vulnerable (or more) libraries in WRES applications (use gradle task and/or inspect build.gradle vs repositories)
- [x] 2 - Tag a commit as staging to kick off pre-release workflow (Pre-release github action)
- [x] 3 - Nominate a commit that has passed the pre-release workflow (Pre-release github action)
- [x] 4 - Check nominated version passes live external services tests (Results sent via emails)
- [x] 5 - Check nominated version passes full system tests (Results sent via emails)
- [x] 6 - Test candidate with 900 series system tests (WRES Test VM or PC)
- [x] 7 - Test migration from scratch (new database instance) - migration succeeds with no issues
- [x] 8 - Test candidate by doing database migration from previous release db with half of system test data to this release with no issues
- [x] 9 - Test candidate with identified alpha (internal) User Acceptance Tests (-ti)
- [x] 10 - Test candidate as a standalone.
- [x] 11 - Push yml files changed
- [x] 12 - Tag the commits that produced the release that passed everything and was already deployed to -prod
- [x] 13 - Update database documentation in wiki
- [x] 14 - Update user documentation
- [x] 15 - Close GitHub tickets actually completed in this release and move tickets not completed to backlog

I'm going to take a lead on this deployment, but may ask for Evan and/or Arvin's help with user acceptance testing and other aspects of the deployment, since this is my first time through the GitHub process.  I'll create a dependency ticket in a minute.

Hank

**Comment by HankHerr-NOAA** [comment]
Noting here something that will need to go into the VLab ticket for local deployment...

A primary change in this deployment is how the certificates are handled. Specifically, Evan updated certificates in all three environments, and I updated the WRES to have certificates specified via environment variables for #296. A necessary checkbox in the VLab deployment will be the need to make sure the -dev `.env`, which already includes the environment variables, is properly setup to use the latest certificates. We then need to port that over to the staging environment, and, later, the production environment. Checkboxes will need to be added to the VLab ticket.

More on that once I'm past the dependency updates. Thanks,

Hank

**Comment by HankHerr-NOAA** [comment]
I'm going to tag commit 80977b3e4126248585bff3c800c0866a446881ff per Step 2 in the deployment wiki so that the pre-release workflow is kicked off.

Hank

**Comment by HankHerr-NOAA** [comment]
Pre-release workflow progress is here:

https://github.com/NOAA-OWP/wres/actions/runs/10851964555

Hank

**Comment by HankHerr-NOAA** [comment]
I'm waiting on emails confirming 4 and 5.

I'll do 6 in a moment. 

For 7 and 8, the wiki with instructions is out of date.  I see "database-migration-check" in the pre-release steps here:

https://github.com/NOAA-OWP/wres/actions/runs/10851964555

Is that enough to check the boxes?  

Hank

**Comment by HankHerr-NOAA** [comment]
Working through Step 6, there appears to be some OBE stuff in the VLab wiki. The proposed sequence is:

```
cd <SNIP>/Deployed_versions/wres-2023XXXX-yyyyyyy/bin
JAVA_OPTS="-Dwres.useDatabase=true" ./wres cleanDatabase
JAVA_OPTS="-Dwres.useDatabase=true" ./wres execute <SNIP>/deployment_tests/scenario900_long/project_config.xml
JAVA_OPTS="-Dwres.useDatabase=true" ./wres execute <SNIP>/deployment_tests/scenario901_noclean/project_config.xml
JAVA_OPTS="-Dwres.useDatabase=true" ./wres execute <SNIP>/deployment_tests/scenario902_long/project_config.xml
JAVA_OPTS="-Dwres.useDatabase=true" ./wres cleanDatabase
JAVA_OPTS="-Dwres.useDatabase=true" ./wres execute <SNIP>/deployment_tests/scenario910_long/project_config.xml
JAVA_OPTS="-Dwres.useDatabase=true" ./wres execute <SNIP>/deployment_tests/scenario911_long/project_config.xml
JAVA_OPTS="-Dwres.useDatabase=true" ./wres cleanDatabase
JAVA_OPTS="-Dwres.useDatabase=true" ./wres execute <SNIP>/deployment_tests/scenario912_long/project_config.xml
```

The default for `wres` when run that way is to run the jobs in-memory, to the best of my knowledge. Hence, the `cleanDatabase` calls do nothing.  If the objective is to run these against a database, then we need to add some steps for setting up the options for the database. Indeed, I think that is the intent, since the test sequence is intended to take advantage of data already ingested in order to see the impact on run times.  

Let me get the test running for myself and then update the wiki. 

Hank

**Comment by james-d-brown** [comment]
Not when `JAVA_OPTS="-Dwres.useDatabase=true" `, as reported.

**Comment by HankHerr-NOAA** [comment]
Oh, duh. I see that in each command. Totally overlooked it. The wiki also points to other wikis for setting up the Postgres instance. Okay. 

Anyway, I need to step out for lunch, but did just confirm the first clean step. I'm going to run the rest blindly and then go.

Thanks,

Hank

**Comment by epag** [comment]
> For 7 and 8, the wiki with instructions is out of date. I see "database-migration-check" in the pre-release steps here:
> 
> https://github.com/NOAA-OWP/wres/actions/runs/10851964555
> 
> Is that enough to check the boxes?
> 
> Hank

Yes, the github automation does our database migration checks for us now

**Comment by HankHerr-NOAA** [comment]
Thanks, Evan! I checked those two boxes.

The 900 series did not finish, because I didn't press enter at the end. Its running 912 now.

System tests and external services tests pass so I checked the boxes:

#130726-104
#130725-334

Hank

**Comment by HankHerr-NOAA** [comment]
900 series is done. Redacted results are attached (link at bottom). Here are the run times in order and the number of output files generated:

```
sh-4.1$ grep "The function" wres.900series.6.26.log 
2024-09-13T16:19:46.610+0000 4402 [main] INFO Functions - The function 'cleanDatabase' took PT5.973141802S
2024-09-13T16:29:01.008+0000 4847 [main] INFO Functions - The function 'execute' took PT9M6.769186941S
2024-09-13T16:31:20.951+0000 8364 [main] INFO Functions - The function 'execute' took PT2M16.945470747S
2024-09-13T16:36:13.684+0000 9346 [main] INFO Functions - The function 'execute' took PT4M49.708141499S
2024-09-13T16:36:34.653+0000 11404 [main] INFO Functions - The function 'cleanDatabase' took PT17.174014581S
2024-09-13T16:44:30.295+0000 11627 [main] INFO Functions - The function 'execute' took PT7M52.949020589S
2024-09-13T16:49:21.108+0000 14823 [main] INFO Functions - The function 'execute' took PT4M47.718652424S
2024-09-13T16:49:40.850+0000 16882 [main] INFO Functions - The function 'cleanDatabase' took PT16.272016394S
2024-09-13T17:25:31.684+0000 26046 [main] INFO Functions - The function 'execute' took PT10M15.959053553S
sh-4.1$ 
sh-4.1$ grep "paths to \/tmp\/wres_evaluation" wres.900series.6.26.log 
2024-09-13T16:29:00.783+0000  [JmsSession [ID:82b422bd-6b95-4afa-8593-cac21b6492a1:2:1] delivery dispatcher] INFO EvaluationConsumer - Consumer sLixKYs5Ym3IBqPTBJ_znxOuH3Q closed evaluation J_3m4y596h21f0mtDX-35tUNT0g, which completed successfully. Wrote 479 paths to /tmp/wres_evaluation_J_3m4y596h21f0mtDX-35tUNT0g.
2024-09-13T16:31:20.717+0000  [JmsSession [ID:819ea729-4fbc-47aa-83b2-b49afc8f0596:2:1] delivery dispatcher] INFO EvaluationConsumer - Consumer yrE2NlwnxFtz4p_OEn30XssxjiQ closed evaluation S8rRzGNlwRu51Kx2NU7cfDuy-io, which completed successfully. Wrote 479 paths to /tmp/wres_evaluation_S8rRzGNlwRu51Kx2NU7cfDuy-io.
2024-09-13T16:36:13.396+0000  [JmsSession [ID:3364ab7c-6e9f-4602-af2e-1411c72c6e6a:2:1] delivery dispatcher] INFO EvaluationConsumer - Consumer FvVIB7gLj1iqh_3orttLVxbmrhI closed evaluation kRGLcA5bMRCpuGJN2939YPJZNy0, which completed successfully. Wrote 479 paths to /tmp/wres_evaluation_kRGLcA5bMRCpuGJN2939YPJZNy0.
2024-09-13T16:44:30.075+0000  [JmsSession [ID:fc2fd0de-a49c-4bc4-aeb4-460ea0fd5e81:2:1] delivery dispatcher] INFO EvaluationConsumer - Consumer uLRE4ahWD5FcM9TTGZByrOtj_G8 closed evaluation r54-5G_GM8e751xLCVo6fXdXiko, which completed successfully. Wrote 479 paths to /tmp/wres_evaluation_r54-5G_GM8e751xLCVo6fXdXiko.
2024-09-13T16:49:20.882+0000  [JmsSession [ID:2f849763-8e9b-48b5-8467-768fe7150279:2:1] delivery dispatcher] INFO EvaluationConsumer - Consumer hzyGhanvOCdHJ2mYDWcvewTMlK8 closed evaluation -bO3-LFuc316GV9WboNDXhuV1JM, which completed successfully. Wrote 479 paths to /tmp/wres_evaluation_-bO3-LFuc316GV9WboNDXhuV1JM.
2024-09-13T17:25:31.045+0000  [JmsSession [ID:987eb3b3-0b29-48c0-8826-76a93f01bfbb:2:1] delivery dispatcher] INFO EvaluationConsumer - Consumer Jh4SuCnb_QRVrOlekT415MQQQo0 closed evaluation rigZB1ScAPJvnjoCo7Dj7j-J_k0, which completed successfully. Wrote 485 paths to /tmp/wres_evaluation_rigZB1ScAPJvnjoCo7Dj7j-J_k0.
```

I don't know how the run times compare, but note that the series was run on the ISED -dev machine using the -dev Postgres database which has in the past been slower than staging and production.

Thanks,

Hank

[wres.900series.6.26.log.gz](https://github.com/user-attachments/files/16997534/wres.900series.6.26.log.gz)

**Comment by HankHerr-NOAA** [comment]
Putting this on hold. Next few steps happen at the NWC, which mean using VLab #134694.

Hank

**Comment by HankHerr-NOAA** [comment]
Working on UAT using the COWRES staging environment. There are 5 tickets in UAT state according to the project:

https://github.com/orgs/NOAA-OWP/projects/38

Hank

**Comment by HankHerr-NOAA** [comment]
Ticket #36 is not ready for deployment at this time.  That leaves a couple of bug fixes and the changes to using environment variables for cert requests. I think I'm going to continue with the deployment, thought there won't be much to advertise in the News items.  

If anyone disagrees, let me know,

Hank

**Comment by HankHerr-NOAA** [comment]
UAT is wrapped up. Now checking out some production evaluations,

Hank

**Comment by HankHerr-NOAA** [comment]
That's tracked in the VLab ticket so putting this on-hold,

Hank

**Comment by HankHerr-NOAA** [comment]
The Push YAML changes checkbox has no instructions in the wiki. Let me check the previous deployment tickets,

Hank

**Comment by HankHerr-NOAA** [comment]
So I know its just a matter of updating the compose entry and workers files. Okay. I  can do that by creating a branch, committing, pushing, and then doing a pull request. 

However, I'm looking for confirmation on how to tag that branch 6.26-docker or something like that.

Hank

**Comment by HankHerr-NOAA** [comment]
In the releases, the naming convention is inconsistent:

https://github.com/NOAA-OWP/wres/releases

Note "Release 6.25-docker" and "Docker Release 6.24".  But, moreover, how do I let the actions know that I need the zip and .tar.gz files noted for both of those releases?

Hank

**Comment by HankHerr-NOAA** [comment]
Yamls pushed and merged in #317. Onto the next step.

Hank

**Comment by HankHerr-NOAA** [comment]
For Step 11:

[hank.herr@nwcal-ised-dev1 wres]$ git tag -a v6.26 80977b3e4126248585bff3c800c0866a446881ff  -m "release 6.26"
[hank.herr@nwcal-ised-dev1 wres]$ git tag -a v6.26-docker 5535de7cd08f0c76f00a61171abfebb42cd1c4a6 -m "compose updates for release v6.26" 

I hope that's right. We'll see soon.

Hank

**Comment by HankHerr-NOAA** [comment]
Forgot the pushing:

```
[hank.herr@nwcal-ised-dev1 wres]$ git push origin v6.26
Counting objects: 1, done.
Writing objects: 100% (1/1), 166 bytes | 0 bytes/s, done.
Total 1 (delta 0), reused 0 (delta 0)
remote: To https://8f7feb5a6fd2988a4e9a3e2ad9175aa9b1a57006@github.com/NOAA-OWP/wres.git
 * [new tag]         v6.26 -> v6.26
[hank.herr@nwcal-ised-dev1 wres]$ git push origin v6.26-docker
Counting objects: 1, done.
Writing objects: 100% (1/1), 185 bytes | 0 bytes/s, done.
Total 1 (delta 0), reused 0 (delta 0)
remote: To https://8f7feb5a6fd2988a4e9a3e2ad9175aa9b1a57006@github.com/NOAA-OWP/wres.git
 * [new tag]         v6.26-docker -> v6.26-docker
 * ```

Waiting for the new releases to be created,
Hank

**Comment by HankHerr-NOAA** [comment]
Looks good:

https://github.com/NOAA-OWP/wres/releases

Hank

**Comment by HankHerr-NOAA** [comment]
No database changes made, I think.

The only change that would have required documentation is the `features` `offset`, but we are slipping that to 6.27. No documentation updates required for this one.

Hank

**Comment by HankHerr-NOAA** [comment]
13 belongs in VLab. I'm going to note that in that ticket and then we should update the wikis afterwards to clean that up. Again. the ITSG document is specifically for our deployment at the NWC supported by ITSG; it has no relation to this ticket.  At least, not without significant changes to the document to untether it from the NWC environment.

Hank

**Comment by HankHerr-NOAA** [comment]
Before I close this ticket, I'm going to update the deployment wiki to match what was done here and in the VLab ticket.

Hank

**Comment by HankHerr-NOAA** [comment]
The only ticket remaining open for this one is the user support ticket for Taylor, as well as this ticket.

Did we discuss how to handle user support tickets in GitHub? Specifically, you want the version employed for that ticket tracked, hence it makes sense to assign it to 6.26. But then we can't close the release with user support tickets hanging on.  In VLab we have a separate user support project to track tickets independently from development. 

Thoughts?

Hank

**Comment by HankHerr-NOAA** [comment]
I removed both branches associated with 6.26 deployment.

Hank

**Comment by epag** [comment]
> The only ticket remaining open for this one is the user support ticket for Taylor, as well as this ticket.
> 
> Did we discuss how to handle user support tickets in GitHub? Specifically, you want the version employed for that ticket tracked, hence it makes sense to assign it to 6.26. But then we can't close the release with user support tickets hanging on. In VLab we have a separate user support project to track tickets independently from development.
> 
> Thoughts?
> 
> Hank

We could tag the PR that put the code in to the release and the un-tag the user support ticket till the user feels it is finished

**Comment by HankHerr-NOAA** [comment]
In this case, there is no associated code. A question was asked and James provided help, but the reporter has not responded yet. So there would be no associated PR. However, we could still untag it I guess... if I understand what you mean by tag. By tag, I assume you mean its placement in the 6.26 milestone?

Hank

**Comment by james-d-brown** [comment]
This is done, right?

**Comment by HankHerr-NOAA** [comment]
This work was done, but there is an open question about what to do with the one user support ticket. I think Evan mentioned untagging the ticket from 6.26 and retagging later when the user replies.

Hank
